src/item.cpp

```diff
@@ -391,6 +392,39 @@ item::item( const recipe *rec, int qty, std::list<item> items, std::vector<item_
     }
 }
 
+item::item( const recipe *rec, item &component )
+    : item( "disassembly", calendar::turn )
+{
+    craft_data_ = cata::make_value<craft_data>();
+    craft_data_->making = rec;
+    std::list<item>items( { component } );
+    components = items;
+
+    if( is_food() ) {
```

Comment:
This assumes that, if a combination isn't food, none of its components will be. Is this justified?

Reply:
The goal here is to deal with rot so if the top level item isn't rotting then none of its components should either.  Honestly I am unsure if there even exist food items that can be disassembled so it might be a moot point.